calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,327 @@
+            self.docker_id
+        ])
+        # Clean trailing whitespace (expect a '\n' at least).
+        ip = ip.strip()
+
+        print('Docker-assigned IP was %s' % ip)
+        return ip
+
+    def _delete_docker_interface(self):
+        """Delete the existing veth connecting to the docker bridge."""
+        print('Deleting eth0')
+
+        # Get the PID of the container.
+        pid = check_output([
+            'docker', 'inspect', '-format', '{{ .State.Pid }}',
+            self.docker_id
+        ])
+        # Clean trailing whitespace (expect a '\n' at least).
+        pid = pid.strip()
+
+        # Set up a link to the container's netns.
+        print(check_output(['mkdir', '-p', '/var/run/netns']))
+        netns_file = '/var/run/netns/' + pid
+        if not os.path.isfile(netns_file):
+            print(check_output(['ln', '-s', '/proc/' + pid + '/ns/net',
+                                netns_file]))
+
+        # Reach into the netns and delete the docker-allocated interface.
+        print(check_output(['ip', 'netns', 'exec', pid,
+                            'ip', 'link', 'del', 'eth0']))
+
+        # Clean up after ourselves (don't want to leak netns files)
+        print(check_output(['rm', netns_file]))
+
+    def _configure_profile(self, endpoint):
+        """
+        Configure the calico profile for a pod.
+
+        Currently assumes one pod with each name.
+        """
+        profile_name = self.pod_name
+        calicoctl('profile', 'add', profile_name)
+        pod = self._get_pod_config()
+
+        self._apply_rules(profile_name, pod)
+
+        self._apply_tags(profile_name, pod)
+
+        # Also add the workload to the profile.
+        calicoctl('endpoint', endpoint.endpoint_id,
+                  'profile', 'set', profile_name)
+        print('Finished configuring profile.')
+
+    def _get_pod_ports(self, pod):
+        """
+        Get the list of ports on containers in the Pod.
+
+        :return list ports: the Kubernetes ContainerPort objects for the pod.
+        """
+        ports = []
+        for container in pod['spec']['containers']:
+            try:
+                more_ports = container['ports']
+                print('Adding ports %s' % more_ports)
+                ports.extend(more_ports)
+            except KeyError:
+                pass
+        return ports
+
+    def _get_pod_config(self):
+        pods = self._get_api_path('pods')
+        print('Got pods %s' % pods)
+
+        for pod in pods:
+            print('Processing pod %s' % pod)
+            if pod['metadata']['name'].replace('-', '_') == self.pod_name:
+                this_pod = pod
+                break
+        else:
+            raise KeyError('Pod not found: ' + self.pod_name)
+        print('Got pod data %s' % this_pod)
+        return this_pod
+
+    def _get_api_path(self, path):
+        """
+        Get the list of pods from the Kube API server.
```

Comment:
Docstring for this guy is wrong - this doesn't just get pods, it can get any arbitrary API item.